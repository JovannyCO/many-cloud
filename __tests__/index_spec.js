describe("Index", () => {
  const app = require("../");
  it("app should not be null", () => {
    expect(app).not.toBeNull();
  });

  describe("#integration_list", () => {
    const loaded_integration_list = require("../lib/integrations/list.json");
    it("/lib/integrations/list.json should exist", () => {
      expect(loaded_integration_list).not.toBeNull();
    });

    it("integration_list should exist", () => {
      expect(app.integration_list).not.toBeNull();
    });

    describe("All Integrations Exist", () => {
      for (let i = 0; i < loaded_integration_list.length; i++)
        it(
          'Integration: "' + loaded_integration_list[i] + '" should exist',
          () => {
            expect(app.integration_list).toContainEqual(
              loaded_integration_list[i]
            );
          }
        );
    });
  });

  describe("#integration", () => {
    const loaded_integration_list = require("../lib/integrations/list.json");
    for (let i = 0; i < loaded_integration_list.length; i++)
      it('Can get Integration: "' + loaded_integration_list[i] + '"', () => {
        expect(app.integration(loaded_integration_list[i])).not.toBeNull();
      });
  });
});
