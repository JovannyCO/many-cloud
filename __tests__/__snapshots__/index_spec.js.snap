// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"Missing name argument."`;

exports[` 2`] = `"Integration \\"not_an_integration\\" not found."`;
